marimo/_plugins/ui/_impl/table.py

```diff
@@ -246,20 +246,33 @@ def __init__(
                 "https://github.com/marimo-team/marimo/issues"
             )
 
-        row_limit = _internal_row_limit or TableManager.DEFAULT_ROW_LIMIT
+        if _internal_row_limit is not None:
+            self._row_limit = _internal_row_limit
+        else:
+            self._row_limit = TableManager.DEFAULT_ROW_LIMIT
 
-        total_rows: Union[int, Literal["too_many"]]
-        if _internal_total_rows == "too_many":
-            total_rows = "too_many"
+        if _internal_summary_row_limit is not None:
+            self._summary_row_limit = _internal_summary_row_limit
         else:
-            total_rows = (
-                _internal_total_rows
-                or self._manager.get_num_rows(force=True)
-                or 0
-            )
-        has_more = total_rows == "too_many" or total_rows > row_limit
-        if has_more:
-            self._filtered_manager = self._filtered_manager.limit(row_limit)
+            self._summary_row_limit = TableManager.DEFAULT_SUMMARY_ROW_LIMIT
+
+        # Holds the data after user searching from original data
+        # (searching operations include query, sort, filter, etc.)
+        self._searched_manager = self._manager
+        # Holds the data after marimo limiting row number from searched data
+        self._limited_manager = self._searched_manager.limit(self._row_limit)
+        # Holds the data after user selecting from the component
+        self._selected_manager: Optional[TableManager[Any]] = None
+
+        # We will need this when calling table manager's to_data()
+        self._format_mapping = format_mapping
+
+        if _internal_total_rows is not None:
+            total_rows = _internal_total_rows
+        else:
+            total_rows = self._manager.get_num_rows(force=True) or 0
```

Comment:
The original logic is quite complex. Simply checking for `is None` should be fine, as the `_internal_total_rows` are only used in the following situations:

1. `marimo._sql` ([link](https://github.com/marimo-team/marimo/blob/60b3e5e5c3159a7bfa48fb563cc69f038dd5ff79/marimo/_sql/sql.py#L93)), where the value will be `"too_many"` or `None`
2. `test_table.py` ([link](https://github.com/marimo-team/marimo/blob/main/tests/_plugins/ui/_impl/test_table.py)), where the value should always be non-zero